File: 2ºCARRERA/tprog/practica2_872815_873338/java/Swap.java
/*
 *  Asignatura : Tecnología de la Programación
 *  Práctica 2:
 *  Autores : Mateo Lorente, Diego  --- 873338
 *            Solana Melero, Carlos --- 872815
 *  Fichero : Swap.java
 */

import java.util.Stack;

public class Swap implements Instruccion {
    public int ejecuta(Stack<Integer> pila, int pc) {
        int num = pila.pop();
        int num2 = pila.pop();
        pila.push(num);
        pila.push(num2);
        pc++;
        return pc;
    }

    public void muestra() {
        System.out.print("swap");
    }
}

File: 2ºCARRERA/tprog/practica2_872815_873338/c++/swap.cc
/*
 *  Asignatura : Tecnología de la Programación
 *  Práctica 2:
 *  Autores : Mateo Lorente, Diego  --- 873338
 *            Solana Melero, Carlos --- 872815
 *  Fichero : swap.cc
 */

#include "swap.h"

void Swap::ejecuta(stack<int32_t> &pila, int& pc) {
    int32_t num = pila.top();
    pila.pop();
    int32_t num2 = pila.top();
    pila.pop();
    pila.push(num);
    pila.push(num2);
    pc++;
}

void Swap::muestra(){
    cout << "swap";
}